### PR TITLE
DSD-542: Card full-click feature

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-542](https://jira.nypl.org/browse/DSD-542)

## This PR does the following:
- Uses Chakra's `LinkBox` and `LinkOverlay` to make an entire `Card` component clickable.
- To enable it, the `mainActionLink` and `mainAriaLabel` props should be passed. (Note, I renamed `masterMainLink` to `mainActionLink`.
- Note: this is pointing to an existing PR until that PR is merged into `development`.

## How has this been tested?

Storybook.

### Accessbility:

I used the combination of `LinkBox` and `LinkOverlay` but we need discernable text in the link in order to pass accessibility tests. The `mainAriaLabel` prop helps achieve this by not rendering text and adding it in the `aria-label` attribute.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

File: src/components/Card/Card.tsx [hunk 1]
```diff
@@ -114,6 +127,39 @@ export function CardActions(props: React.PropsWithChildren<CardActionsProps>) {
...
+  }
+
+  return mainActionLink ? (
+    <ChakraLinkBox>
+      <ChakraLinkOverlay
+        href={mainActionLink}
+        _before={zIndex}
+        aria-label={mainAriaLabel}
+      />
+      {children}
```
Comment (bigfishdesign13) [inline src/components/Card/Card.tsx]: I believe wrapping everything in the `Card` in the `LinkOverlay` does not allow the internal buttons and links to be mouse clickable unless you tab to them.  What do you think about only wrapping `cardContent` in `ChakraLinkOverlay`?  That would align with the example on the Chakra site where the only heading is wrapped and the buttons are outside.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
I'd like to get the internal buttons and links to be mouse clickable.  Please see my question/comment.

**Comment by bigfishdesign13** [comment]
@EdwinGuzman Sorry, I forgot to mention that the `Latest Version` number also needs to be ticked up.

File: src/components/Card/Card.tsx (hunk 1 above)
Comment (EdwinGuzman) [reply src/components/Card/Card.tsx]: Yeah, I'll try that. It makes more sense.

**Comment by EdwinGuzman** [comment]
@bigfishdesign13 updated. Added more information in a section about this and the headings as well. This mostly came up because the Chakra's `LinkOverlay` component should wrap text in the first `CardHeading` component it should find if there are more than one.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr742-3whcdcgrc9sdfktw1zg7gt3wy7hjyq7n.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/617afe96cb7f1e87cbdbaeb6

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr742-3whcdcgrc9sdfktw1zg7gt3wy7hjyq7n.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/617afe96cb7f1e87cbdbaeb6

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr742-3whcdcgrc9sdfktw1zg7gt3wy7hjyq7n.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/617afe96cb7f1e87cbdbaeb6

**Review by bigfishdesign13 — APPROVED** [review]
Works great.

**Pull request merged**